### ISSUE TITLE
Sik-Ho Tang | Review -- Striving for Simplicity: The All Convolutional Net.

### ISSUE THREAD

**Issue by NorbertZheng** [NorbertZheng/read-papers#97]
Sik-Ho Tang. [Review — Striving for Simplicity: The All Convolutional Net](https://sh-tsang.medium.com/review-striving-for-simplicity-the-all-convolutional-net-2b2f762c1542).

**Comment by NorbertZheng** [comment]
## Overview
Striving for Simplicity: The All Convolutional Net. All-CNN, by University of Freiburg. 2015 ICLR Workshop, Over 3800 Citations.

Convolutional Neural Network, CNN, Image Classification.

 - CNNs are commonly composed of alternative convolution and max-pooling layers followed by a small number of fully connected layers.
 - In this paper, **max-pooling can simply be replaced by a convolutional layer with increased stride without loss in accuracy**.
 - **Fully connected layers are replaced by global average pooling layer**.

**Comment by NorbertZheng** [comment]
## Proposed All-CNN
### Two Means to Replace Pooling
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/21d61cc0-b6f2-4781-a078-1dab9608fd94)
Convolution With Stride Larger Than 1 to Reduce Spatial Size More Aggressively (Figure from [here](https://adeshpande3.github.io/A-Beginner%27s-Guide-To-Understanding-Convolutional-Neural-Networks-Part-2/)).

There are two means suggested to replace pooling for **spatial dimensionality reduction** (i.e. along width & height axis, instead of filter axis):
 - To increase the stride of the convolutional layer that preceded it. However, it significantly reduces the overlap of the convolutional layer, and results in less accurate recognition.
 - Or to replace the pooling layer by a normal convolution with stride larger than one.

**The second option results in an increase of overall network parameters, yet without loss in accuracy.**

**Comment by NorbertZheng** [comment]
### Global Average Pooling to Replace Fully Connected Layer
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/ba9614a6-3911-400c-aef9-e1e70c445fca)
Global Average Pooling in [NIN](https://towardsdatascience.com/review-nin-network-in-network-image-classification-69e271e499ee) to Replace Fully Connected Layer (Image from [NIN](https://towardsdatascience.com/review-nin-network-in-network-image-classification-69e271e499ee)).

 - This is firstly suggested in [NIN](https://towardsdatascience.com/review-nin-network-in-network-image-classification-69e271e499ee).

**Using global average pooling to replace fully connected layer helps to reduce large amount of parameters.**

**Comment by NorbertZheng** [comment]
### Three Base Models
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/9370074c-6172-4a88-92fe-c1ad061f9a2f)
The three base networks used for classification on CIFAR-10 and CIFAR-100.

Overall, three base models are suggested which consist only of convolutional layers with rectified linear non-linearities and an averaging + softmax layer to produce predictions over the whole image.

**Comment by NorbertZheng** [comment]
### Three Derived Models from Base Models
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/a7796cc4-c3c6-43ca-afaa-4e01d63a2f53)
Model description of the three networks derived from base model C used for evaluating the importance of pooling in case of classification on CIFAR-10 and CIFAR-100.

Further enhanced models are derived from base models. The derived models for base models A and B are built analogously but not shown in the above table.

5×5 convolutions are replaced by 2 consecutive 3×3 convolutions.

**Comment by NorbertZheng** [comment]
### Detailed Architecture
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/16ffc4be-97b9-4e58-8355-663b14d67592)
Architecture of the Large All-CNN network for CIFAR-10.

The above shows the detailed architecture for CIFAR-10.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/aa950137-8ce3-477d-85d3-7dfdfca333f6)
Architecture of the ImageNet network.

The above shows the detailed architecture for ImageNet.

**Comment by NorbertZheng** [comment]
## Experimental Results
### Ablation Study
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/b24d151d-5675-4eb2-a277-73f5a6444231)
All-CNN-C has the best performance.

### SOTA Comparison
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/8d6cbad8-0fe9-4159-bd84-a936dda23914)
Test error on CIFAR-10 and CIFAR-100 for the All-CNN compared to the state of the art from the literature.

On CIFAR-10, All-CNN is the All-CNN-C. It outperforms [Maxout](https://towardsdatascience.com/review-maxout-network-image-classification-40ecd77f7ce4?source=post_page---------------------------), and [NIN](https://towardsdatascience.com/review-nin-network-in-network-image-classification-69e271e499ee), etc. On CIFAR-100, All-CNN-C obtains competitive performance.

### ImageNet
An upscaled version of the All-CNN-B network is trained, which has 12 convolutional layers.

This network achieves a Top-1 validation error of 41.2% on ILSVRC-2012, when only evaluating on the center 224×224 patch, — which is comparable to the 40.7% Top-1 error reported by [AlexNet](https://medium.com/coinmonks/paper-review-of-alexnet-caffenet-winner-in-ilsvrc-2012-image-classification-b93598314160?source=post_page---------------------------).

**Comment by NorbertZheng** [comment]
(There are also sections to visualize the feature map response using deconvolutions which are something similar to [ZFNet](https://medium.com/coinmonks/paper-review-of-zfnet-the-winner-of-ilsvlc-2013-image-classification-d1a5a0c45103), please feel free to read the paper.)

**Comment by NorbertZheng** [comment]
## Reference
 - [2015 ICLR Workshop] [All-CNN] [Striving for Simplicity: The All Convolutional Net](https://arxiv.org/abs/1412.6806).
